transmogrifier/sources/xml/datacite.py

```diff
@@ -302,20 +248,113 @@ def get_optional_fields(self, xml: Tag) -> dict | None:
+        alternate_titles = []
+        alternate_titles.extend(
+            [
+                timdex.AlternateTitle(
+                    value=str(title.string.strip()),
+                    kind=title["titleType"],
+                )
+                for title in source_record.find_all("title", string=True)
+                if title.get("titleType")
+            ]
+        )
+        alternate_titles.extend(list(cls._get_additional_titles(source_record)))
+        return alternate_titles
+
+    @classmethod
+    def _get_additional_titles(
+        cls, source_record: Tag
+    ) -> Iterator[timdex.AlternateTitle]:
+        """Get additional titles from get_main_titles method."""
+        for index, title in enumerate(cls.get_main_titles(source_record)):
+            if index > 0:
+                yield timdex.AlternateTitle(value=title)
+
+    @classmethod
+    def get_content_type(cls, source_record: Tag, source_record_id: str) -> list[str]:
+        content_types = []
+        if resource_type := source_record.metadata.find("resourceType"):
+            if content_type := resource_type.get("resourceTypeGeneral"):
+                if cls.valid_content_types([content_type]):
+                    content_types.append(str(content_type))
+                else:
+                    message = f'Record skipped based on content type: "{content_type}"'
+                    raise SkippedRecordEvent(message, source_record_id)
+        else:
+            logger.warning(
+                "Datacite record %s missing required Datacite field resourceType",
+                source_record_id,
+            )
+        return content_types
+
+    @classmethod
+    def get_contributors(cls, source_record: Tag) -> list[timdex.Contributor]:
+        contributors = []
+        contributors.extend(list(cls._get_creators(source_record)))
+        contributors.extend(list(cls._get_contributors(source_record)))
+        return contributors
+
+    @classmethod
+    def _get_creators(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
+        for creator in source_record.metadata.find_all("creator"):
+            if creator_name := creator.find("creatorName", string=True):
+                yield timdex.Contributor(
+                    value=str(creator_name.string),
+                    affiliation=[
+                        str(affiliation.string)
+                        for affiliation in creator.find_all("affiliation", string=True)
+                    ]
+                    or None,
+                    identifier=[
+                        cls.generate_name_identifier_url(name_identifier)
+                        for name_identifier in creator.find_all(
+                            "nameIdentifier", string=True
+                        )
+                    ]
+                    or None,
+                    kind="Creator",
+                )
+
+    @classmethod
+    def _get_contributors(cls, source_record: Tag) -> Iterator[timdex.Contributor]:
```

Comment:
Would a docstring help here to clarify it's accessing the source record contributor?

Reply:
It's a good thought, but I don't think so.  

I would expect patterns like `get_dates()` calls `_get_published_dates()`, `get_subject_dates()`, etc.  This one is tricky because it's quite literally "contributors" in the datacite XML, so I think it's accurate.

Probably just a convention / norm, where these private helpers really aren't meant to be quickly scanned, but dug into when understanding a higher level field method.